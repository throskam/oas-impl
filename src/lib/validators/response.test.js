import ResponseValidator from './response'

describe('Empty', () => {
  const response = {}

  const validator = new ResponseValidator(response)

  it('should return no errors', () => {
    expect(validator.validate()).toMatchSnapshot()
  })
})

describe('Headers & Content', () => {
  const response = {
    headers: {
      'x-foo-bar': {
        schema: {
          type: 'integer'
        }
      }
    },
    content: {
      'application/json': {
        schema: {
          type: 'integer'
        }
      }
    }
  }

  const validator = new ResponseValidator(response)

  it('should return no errors when the headers and content are valid', () => {
    const payload = { header: { 'x-foo-bar': 1 }, content: 1, mediaType: 'application/json' }
    expect(validator.validate(payload)).toMatchSnapshot()
  })

  it('should return validation errors when the headers and content are invalid', () => {
    const payload = { header: { 'x-foo-bar': 'string', content: true, mediaType: 'application/json' } }
    expect(validator.validate(payload)).toMatchSnapshot()
  })
})
