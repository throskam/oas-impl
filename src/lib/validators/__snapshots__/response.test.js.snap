// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should return no errors 1`] = `Array []`;

exports[`Headers & Content should return no errors when the headers and content are valid 1`] = `Array []`;

exports[`Headers & Content should return validation errors when the headers and content are invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "response.headers.header[x-foo-bar].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
  Object {
    "path": "response.content",
    "rule": "content-media-type-required",
    "value": undefined,
  },
]
`;
